### PR TITLE
Add test for git harvester

### LINKED ISSUE #113: Test git harvesting
Needs monkeypatching `subprocess.run`ning the git log --pretty call and matching the harvesting context contents.

### PR DESCRIPTION
**Pull request by sdruskat**
- Fix #113
- Assert the JSONized harvest context after running the git harvester

### CODE REVIEW

File: test/hermes_test/commands/harvest/test_git.py
Comment (zyzzyxdonta) [inline test/hermes_test/commands/harvest/test_git.py:107]: Is `MagicMock` necessary here? It seems to work just fine with `Mock`.

File: test/hermes_test/commands/harvest/test_git.py
Comment (jkelling) [inline test/hermes_test/commands/harvest/test_git.py:120]: `subprocess.run` is called two times. The mock only provides valid output for the second call. The first call
https://github.com/hermes-hmc/workflow/blob/f02ff2a2b7a9ea2b2dd278fd9c0c0b827ac41bbc/src/hermes/commands/harvest/git.py#L287
is to get the current branch name/state. If the harvester would do sanity checks on the git output, `harvest_git` would be expected to fail here. This information is also passed to the context update
https://github.com/hermes-hmc/workflow/blob/f02ff2a2b7a9ea2b2dd278fd9c0c0b827ac41bbc/src/hermes/commands/harvest/git.py#L315-L323
.. but apparently does not end up in the context json, otherwise the test would fail ... what is going on here?

**Review by jkelling — CHANGES_REQUESTED** [review]

File: test/hermes_test/commands/harvest/test_git.py
Comment (jkelling) [reply test/hermes_test/commands/harvest/test_git.py:120]: The branch is passed to `HermesHarvestContext.updat_from` as a keyword arg and remian unused therein. I.e. the proposed mock works *until* we decide to record the branch in the harvest context.

Comment (sdruskat) [reply test/hermes_test/commands/harvest/test_git.py:120]: > This information is also passed to the context update
> 
> https://github.com/hermes-hmc/workflow/blob/f02ff2a2b7a9ea2b2dd278fd9c0c0b827ac41bbc/src/hermes/commands/harvest/git.py#L315-L323
> 
> .. but apparently does not end up in the context json, otherwise the test would fail ... what is going on here?

Huh, must've slipped through review. The information *isn't* actually passed to the context. To pass it, either include it in the `dict` itself

```python
    ctx.update_from({ 
     '@context': [ 
         "https://doi.org/10.5063/schema/codemeta-2.0", 
         {'hermes': 'https://software-metadata.pub/ns/hermes/'} 
     ], 
  
     '@type': "SoftwareSourceCode", 
     'contributor': [contributor.to_codemeta() for contributor in git_contributors._all], 
     'branch': git_branch
 }) 
```

Or add it separately (which I'd prefer for reasons of readability) with `ctx.update('branch', git_branch)`.
This is a bug, tracked in

**Review by zyzzyxdonta — APPROVED** [review]
👍🏻

**Comment by sdruskat** [comment]
Waiting for @jkelling's re-review before merging.

**Review by jkelling — APPROVED** [review]

**Pull request merged**
